ArdusatSDK.cpp

```diff
@@ -453,266 +257,1212 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
+  _resetOutBuf();
 
-  dtostrf(input.y, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
+  sprintf(nameBuf, labelA, sensorName);
+  _writeJSONValue(&_getOutBuf()[_output_buf_len], nameBuf, unit_to_str(unit), valueA);
 
-  dtostrf(input.z, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
+  sprintf(nameBuf, labelB, sensorName);
+  _writeJSONValue(&_getOutBuf()[_output_buf_len], nameBuf, unit_to_str(unit), valueB);
 
-  _add_checksum_to_csv_buffer(sensorName, 3, input.x, input.y, input.z);
-  _getOutBuf()[_output_buf_len++] = '\n';
+  sprintf(nameBuf, labelC, sensorName);
+  _writeJSONValue(&_getOutBuf()[_output_buf_len], nameBuf, unit_to_str(unit), valueC);
   return _getOutBuf();
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(magnetic)
-const char * magneticToCSV(const char *sensorName, magnetic_t & input) {
-  _headerToCSV(&(input.header), sensorName);
-
-  dtostrf(input.x, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
-
-  dtostrf(input.y, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _getOutBuf()[_output_buf_len++] = CSV_SEPARATOR;
-
-  dtostrf(input.z, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-
-  _add_checksum_to_csv_buffer(sensorName, 3, input.x, input.y, input.z);
-  _getOutBuf()[_output_buf_len++] = '\n';
-
+/**
+ * Creates a JSON string with the appropriate values
+ *
+ * @param sensor_name the label to be given to some data
+ * @param unit the unit of measurement used
+ * @param value the sensor value
+ *
+ * @return the output buffer
+ */
+const char * valueToJSON(const char *sensorName, unsigned char unit, float value) {
+  _resetOutBuf();
+  _writeJSONValue(_getOutBuf(), sensorName, unit_to_str(unit), value);
   return _getOutBuf();
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(temperature)
-const char * temperatureToCSV(const char *sensorName, temperature_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
-
-  dtostrf(input.t, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-
-  _add_checksum_to_csv_buffer(sensorName, 1, input.t);
-  _getOutBuf()[_output_buf_len++] = '\n';
 
-  return _getOutBuf();
+/**************************************************************************//**
+ * @brief   Constructs Acceleration sensor object
+ * @ingroup acceleration
+ *
+ * Example Usage:
+ * @code
+ *     Acceleration accel = Acceleration(); // Instantiate sensor object
+ *     accel.begin();                       // Initialize sensor
+ *     Serial.println(accel.readToJSON());  // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+Acceleration::Acceleration(void) {
+  Acceleration::sensorId = SENSORID_ADAFRUIT9DOFIMU;
```

Comment:
Typically in C++ we use static initializers rather than a constructor method body to initialize member variables that don't involve complicated computations.